velox/connectors/hive/SplitReader.h

```diff
@@ -19,6 +19,7 @@
 #include "velox/common/base/RandomUtil.h"
 #include "velox/connectors/hive/FileHandle.h"
 #include "velox/dwio/common/Options.h"
+#include "velox/dwio/common/Reader.h"
```

Comment:
Why this change?

Reply:
I reverted this. It was added to fix a build failure from the cleanup. I included the header in the cpp file.

Reply:
In my latest rebase, I decided to add this back. SplitReader has the following two fields
```
 std::unique_ptr<dwio::common::Reader> baseReader_;
 std::unique_ptr<dwio::common::RowReader> baseRowReader_;
```
Any class (eg: HiveConnector.cpp) that creates an instance of SplitReader needs to include `#include "velox/dwio/common/Reader.h"` to avoid the error below. It is odd to impose this requirement on the client of 
SplitReader.

```
In file included from /opt/rh/gcc-toolset-12/root/usr/include/c++/12/memory:76,
                 from /usr/local/include/folly/Function.h:205,
                 from /usr/local/include/folly/CancellationToken.h:20,
                 from /__w/velox/velox/./velox/connectors/Connector.h:18,
                 from /__w/velox/velox/./velox/connectors/hive/HiveConnector.h:18,
                 from /__w/velox/velox/velox/connectors/hive/HiveConnector.cpp:17:
/opt/rh/gcc-toolset-12/root/usr/include/c++/12/bits/unique_ptr.h: In instantiation of 'void std::default_delete<_Tp>::operator()(_Tp*) const [with _Tp = facebook::velox::dwio::common::Reader]':
/opt/rh/gcc-toolset-12/root/usr/include/c++/12/bits/unique_ptr.h:396:17:   required from 'std::unique_ptr<_Tp, _Dp>::~unique_ptr() [with _Tp = facebook::velox::dwio::common::Reader; _Dp = std::default_delete<facebook::velox::dwio::common::Reader>]'
/__w/velox/velox/./velox/connectors/hive/SplitReader.h:83:11:   required from 'void std::default_delete<_Tp>::operator()(_Tp*) const [with _Tp = facebook::velox::connector::hive::SplitReader]'
/opt/rh/gcc-toolset-12/root/usr/include/c++/12/bits/unique_ptr.h:396:17:   required from 'std::unique_ptr<_Tp, _Dp>::~unique_ptr() [with _Tp = facebook::velox::connector::hive::SplitReader; _Dp = std::default_delete<facebook::velox::connector::hive::SplitReader>]'
/__w/velox/velox/./velox/connectors/hive/HiveDataSource.h:37:7:   required from 'void std::default_delete<_Tp>::operator()(_Tp*) const [with _Tp = facebook::velox::connector::hive::HiveDataSource]'
/opt/rh/gcc-toolset-12/root/usr/include/c++/12/bits/unique_ptr.h:396:17:   required from 'std::unique_ptr<_Tp, _Dp>::~unique_ptr() [with _Tp = facebook::velox::connector::hive::HiveDataSource; _Dp = std::default_delete<facebook::velox::connector::hive::HiveDataSource>]'
/__w/velox/velox/velox/connectors/hive/HiveConnector.cpp:64:42:   required from here
/opt/rh/gcc-toolset-12/root/usr/include/c++/12/bits/unique_ptr.h:93:23: error: invalid application of 'sizeof' to incomplete type 'facebook::velox::dwio::common::Reader'
   93 |         static_assert(sizeof(_Tp)>0,
      |                       ^~~~~~~~~~~
```

Reply:
CI job failed without this change
https://github.com/facebookincubator/velox/actions/runs/10433134539/job/28894546456